File: server_integration_tests/servers/java_server/src/main/java/dev/graphlink/javaserver/impl/CollideServiceImpl.java
package dev.graphlink.javaserver.impl;

import dev.graphlink.javaserver.generated.services.CollideService;
import dev.graphlink.javaserver.generated.types.Collide;
import org.springframework.stereotype.Service;

@Service
public class CollideServiceImpl implements CollideService {

    @Override
    public Collide collide() {
        // The two colliding wire fields (`class` and `class_`) carry distinct
        // values so the round-trip can prove they stayed separate identifiers.
        return Collide.builder().id("1").class_2("A").class2("B").build();
    }
}

File: server_integration_tests/servers/kotlin_server_blocking/src/main/kotlin/dev/graphlink/kotlinserverblocking/impl/CollideServiceImpl.kt
package dev.graphlink.kotlinserverblocking.impl

import dev.graphlink.kotlinserverblocking.generated.services.CollideService
import dev.graphlink.kotlinserverblocking.generated.types.Collide
import org.springframework.stereotype.Service

@Service
class CollideServiceImpl : CollideService {
    // The two colliding wire fields (`class` and `class_`) carry distinct
    // values so the round-trip can prove they stayed separate identifiers.
    override fun collide(): Collide = Collide(id = "1", class_2 = "A", class2 = "B")
}
